sklearn/utils/validation.py

```diff
@@ -487,6 +502,10 @@ def check_array(array, accept_sparse=False, accept_large_sparse=True,
             # list of accepted types.
             dtype = dtype[0]
 
+    if has_pd_interger_array:
+        # If there are any pandas integer extension arrays,
+        array = array.astype(None)
```

Comment:
I think `.astype(np.dtype('float64'))` would be best (assuming you always want 64-bit here).

Reply:
Or `to_numpy()`? That's even more explicit (and certainly given that astype behaviour might still change, eg to always return new ExtensionArray), although only available for more recent versions? (which might not be a problem, since this is still experimental anyway)

Reply:
Looking at the actual code now, `array` here will typically be a DataFrame (in case of pandas), and it seemt that for `DataFrame.to_numpy` we didn't add the `na_value` keyword .. (that's a pity, as I think `df.to_numpy(dtype='float64', na_value=np.nan)` would have been the most robust going forward).